app/controllers/users_controller.rb

```diff
@@ -10,16 +14,43 @@ def create
     end
   end
 
-  def show
-    @user = User.find_by(params[:id])
+  def index
+    @users = User.paginate(page: params[:page])
   end
 
+  def show; end
+
   def new
     @user = User.new
   end
 
+  def edit; end
+
+  def update
+    if @user.update(user_params)
+      flash[:success] = t "users.update.success"
+      redirect_to @user
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    @user.destroy
+    flash[:success] = t "users.destroy.success"
+    redirect_to users_url
+  end
+
   private
 
+  def load_user
+    @user = User.find_by id: params[:id]
+    return if @user
+
+    flash[:danger] = t "users.login.danger"
```

Comment:
```suggestion
    flash[:danger] = t "users.login.danger"
```
chị nghĩ message này là user not found chứ sao lại là user chưa login nhỉ